apple/swift-docc-render
# Cleanup and improvements to the Navigator

Bug/issue #, if applicable: 97716711

## Summary

This PR refactors the Navigator to make it more robust and cleans up the styling.

1. Spacings to be configurable via CSS variables.
2. Cleans up the sticky styling so its more robust
3. Allows specifying the filter positioning via a prop
4. Allow having none-truncated titles in nav items

## Dependencies

NA

## Testing

Steps:
1. Make sure there are no styling regressions.
1. Try to switch the filter positioning in the Vue dev tools by flipping the boolean prop on the `Navigator.vue`
2. Try to scroll fast and see if there are perf issues
3. Try in devtools to remove the `whitespace: nowrap` so titles overflow. Assert the navigator works as expected.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
